jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -105,6 +105,9 @@ def _dataset_selected_changed(self, event={}):
             if telescope == 'JWST':
                 if 'photometry' in meta and 'pixelarea_arcsecsq' in meta['photometry']:
                     self.pixel_area = meta['photometry']['pixelarea_arcsecsq']
+                    if 'bunit_data' in meta and meta['bunit_data'] == u.Unit("MJy/sr"):
+                        # Hardcode the flux conversion factor from MJy to ABmag
+                        self.flux_scaling = 3631e-6

```

Comment:
I was thinking more like this. And `self._mag_unit = u.mag` for everything else. So when the code sets the `'aperture_sum_mag'`, it grabs whatever is set in `self._mag_unit`. Does that make sense?
```suggestion
                        self.flux_scaling = 3631e-6
                        self._mag_unit = u.ABmag
```

Reply:
That causes the following test failure:
```
        assert_quantity_allclose(phot_plugin.flux_scaling, 3.631e-3, rtol=1e-3)
        phot_plugin.flux_scaling = 1  # Simple mag, no zeropoint
        phot_plugin.vue_do_aper_phot()
        tbl = imviz_helper.get_aperture_photometry_results()
>       assert_quantity_allclose(tbl[0]['xcentroid'], 970.935492 * u.pix)
E       TypeError: 'NoneType' object is not subscriptable

test_parser.py:277: TypeError
======================================================================== short test summary info =========================================================================
FAILED test_parser.py::TestParseImage::test_parse_jwst_nircam_level2 - TypeError: 'NoneType' object is not subscriptable
```

Reply:
When `tbl` is `None`, that means something went wrong. Try to raise the exception here:

https://github.com/spacetelescope/jdaviz/blob/c523cf6897212b905411c6681639e5b2f5baf776/jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py#L413

And run the test locally and see the traceback. 🤞

Reply:
`Aperture photometry failed: TypeError("unsupported operand type(s) for *: 'Quantity' and 'MagUnit'")` For some reason `u.mag` can be used just fine but `u.ABmag` causes this error. I see there is an [astropy ticket](https://github.com/astropy/astropy/issues/12472) for confusion with the mag system...soooo I'll leave this PR as is for now to move on to other work.

Reply:
Ah, well. We tried.

Maybe add a sentence in the paragraph mentioning `flux_scaling` in `docs/imviz/plugins.rst` to explain that it is autopopulated in this case and user should expect AB magnitude when that happens?

Reply:
Done!